src/interpret.jl

```diff
@@ -474,11 +481,16 @@ by normal dispatch, whereas a vector of `JuliaStackFrame`s will use recursive in
 """
 function step_expr!(stack, frame, istoplevel::Bool=false)
     pc = _step_expr!(stack, frame, frame.pc[], istoplevel)
-    pc === nothing && return nothing
+    (pc === nothing || isa(pc, BreakpointRef)) && return pc
     frame.pc[] = pc
 end
 
-function handle_err(frame, err)
+function handle_err(stack, frame, pc, err)
+    if break_on_error[]
```

Comment:
Hm, should we really break here even if the error end up getting caught?

Reply:
Interesting question. Should we support both? `try/catch` often make debugging hard, breaking on the initial error can help figure out what's going wrong. But you're also right that when this is used as semi-normal control flow, the current setting is problematic.

If you like the idea of supporting both, any suggestions for names? How about `:none` (equivalent to `false` now), `:uncaught`, `:caught` (corresponding to `true` now)?

Reply:
Having both options sounds great, and the names are fine imho.